src/system_tests/e2e/base_test.py

```
@@ -123,10 +156,24 @@ async def wait_for_server(self, port: int, max_retries: int = 250, retry_interva
+                    # Check process status one last time
+                    if process is not None and process.poll() is not None:
+                        error_msg += f"\n{process_name} process died with return code {process.returncode}"
+                        if log_file and os.path.exists(log_file):
+                            try:
+                                with open(log_file, 'r') as f:
+                                    log_content = f.read()
+                                    if log_content:
+                                        log_lines = log_content.split('\n')
+                                        last_lines = '\n'.join(log_lines[-50:])
+                                        error_msg += f"\n\nLast 50 lines of {process_name} log ({log_file}):\n{last_lines}"
+                            except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)